pilot/pkg/proxy/envoy/discovery.go

```diff
@@ -300,6 +308,26 @@ func NewDiscoveryService(ctl model.Controller, configCache model.ConfigStoreCach
 		container.ServeMux.HandleFunc("/debug/pprof/trace", pprof.Trace)
 	}
 	out.Register(container)
+
+	if len(o.WebhookEndpoint) > 0 {
+		out.webhookEndpoint = o.WebhookEndpoint
+		transport := &http.Transport{}
```

Comment:
nit: create the default transport in an else below?

Reply:
dont understand.. the flow is, you create the empty transport. if its Unix domain socket, you set the dial context field in the transport, else leave it as empty.